docs/source/api/v4/cdn_locks.rst

```diff
@@ -71,6 +72,7 @@ Allows user to acquire a lock on a CDN.
 
 :Auth. Required: Yes
 :Roles Required: "admin" or "operations"
+:Permissions Required: CDNLOCK:CREATE
```

Comment:
Should things like this also require `CDN-READ` (or equivalent)? Like, a user is allowed to delete a CDN that they aren't allowed to know exists. But the information is exposed by being able to manipulate the objects anyway, and then if e.g. in this case, if the CDN is already locked and it returns an error like "CDN 'CDN-Name' is already locked" then should it check for Permissions before displaying the CDN name in the error message?

Reply:
By similar routes do you mean those not having the read permission on an endpoint that deletes/updates/creates?

Reply:
Yeah, or related things like locks, capability associations, and snapshots

Reply:
I'll explicitly require the `READ` perm on the 'same' route and directly related routes.

I'm torn on `snapshots`, it would pretty much need read permission for everything which doesn't seem optimal; hence why I have the permissions specifically for snapshots.

Reply:
That's fair, I don't have any better ideas for Snapshots either.

Reply:
I've updated the routes. Each route requires read permissions for the _required_ fields.